cli/commands/event/delete.go

```diff
@@ -32,7 +32,12 @@ func DeleteCommand(cli *cli.SensuCli) *cobra.Command {
 				}
 			}
 
-			err := cli.Client.DeleteEvent(entity, check)
+			namespace := cli.Config.Namespace()
+			if namespaceFlag, _ := cmd.Flags().GetString("namespace"); namespaceFlag != "" {
+				namespace = namespaceFlag
+			}
```

Comment:
After #2770 is merged, I'm fairly sure those calls to `cmd.Flags().Get*()` will not be needed anymore. They were a crutch to make up for an expected parse error earlier in the program's execution, which #2770 fixes through the error whitelisting facility newer versions of Cobra provide!

`namespace := cli.Config.Namespace()` should now correctly give you the right value: the default namespace, overridden by what's in the config file, in turn overridden by a `--namespace` flag anywhere on the command line.

Not requesting a change, just sharing that fact for future changes to those commands!

Reply:
Yeah, I'll probably rebase after your PR lands, update the necessary things and retest.